src/backend/XmlHttpRequest.ts

```diff
@@ -396,7 +414,11 @@ export default class XmlHttpRequest extends BaseFileSystem implements FileSystem
   private _requestFileAsync(p: string, type: 'json', cb: BFSCallback<any>): void;
   private _requestFileAsync(p: string, type: string, cb: BFSCallback<any>): void;
   private _requestFileAsync(p: string, type: string, cb: BFSCallback<any>): void {
-    asyncDownloadFile(this.getXhrPath(p), type, cb);
+    if (this._useFetch) {
```

Comment:
Oh, I didn't mean for that to sound sassy here. I'm referring to [this comment](https://github.com/jvilk/BrowserFS/pull/185#issuecomment-319266217):

> To avoid redundant conditionals (e.g. if (shouldFetch) all over the code), you can add private fields to the file system that save a reference to the chosen method of fetching / getting the size of files at file system construction time.